core/src/main/java/io/kestra/plugin/core/condition/DateTimeBetween.java

```diff
@@ -102,18 +101,19 @@ public class DateTimeBetween extends Condition implements ScheduleCondition {
     @Override
     public boolean test(ConditionContext conditionContext) throws InternalException {
         Map<String, Object> vars = conditionContext.getVariables();
-        String render = conditionContext.getRunContext().render(date, vars);
-        ZonedDateTime currentDate = DateUtils.parseZonedDateTime(render);
+        String render = conditionContext.getRunContext().render(date).as(String.class, vars).orElseThrow();
+        
+        ZonedDateTime currentDate = (render != null) ? DateUtils.parseZonedDateTime(render) : ZonedDateTime.now();
```

Comment:
```suggestion
        ZonedDateTime currentDate = conditionContext.getRunContext().render(date).as(String.class, vars).map(it -> DateUtils.parseZonedDateTime(it)).orElseThrow();
```

`render` cannot be null!

Thi

Reply:
I am getting an API deprecated issue for the suggested code when running the test locally:

<img width="1244" height="134" alt="image" src="https://github.com/user-attachments/assets/4cfec56c-db16-490b-9b8e-6afe06582804" />